### DISCUSSION TITLE
add controller support yes/no for games

### DISCUSSION THREAD

**Discussion by nsklaus** [ThatGuySam/doesitarm#597] [Ideas]
when submitting games,
it would be nice if we could get details about controller support.

for example,
- batman arkham city, controller support : yes (tested with ds4)
- dying light, controller support: no  (tested with ds4)

would the maintainer of doesitarm.com would allow if i started to fill the game section with data imported from applesilicongames.com with PR, in the same way apps are done ? if we did so, we could have more freedom with the amount of data displayed.
right now, if i wanted to add or modify something i get redirected toward applesilicongames.com submission form.
but this form have limited / fixed amount of things one can report about.
like, there's no controller support section in the form. there are also other details that could be worth adding too to each game entry.
for example: 
used 3d api (opengl, metal, direct3d, vulkan) 
environment (crossover, parallels, rosetta2, native)
controller: ds4, xbox, steam gamepad ..
game version tested ... (1.0 or 3.3465, .. )

**Comment by ThatGuySam** [comment]
This is an interesting idea. 

We partner with Apple Silicon Games to use their data in exchange for crediting the source so I want to always honor any data they provide us. 

However, if we generate the data ourselves that's fine. 

I could set up the site to pull from a Google Sheet with info from reports. 

How many games were you thinking you would want to report on at first?

**Comment by nsklaus** [comment]
initialy i planned to import the whole from ASG (applesilicongames), working my way one title at a time.
then once it's all in, the idea was that we could be updating each entries when there was some new info to share.
for example, today i've found a way to add native controller support in dying light, and alien isolation. 
if doesitarm.com had the games entries the same way it has for apps, then i would have updated these titles.
also, as i go and test games, i would have updated other game too with controller support, and other useful little tweaks and settings.
the goal was not to have each entries filled with a lot of datas, but rather, minimum info needed to know if game can run well, and how to do it. what environment they run on, if they support controller or not, and things like that. basicaly what ASG already provide plus little bits of added bonus here and there. the point was to have them editable on PR by the community so we can all participate when something new is discovered.

also, ASG and AGW (applegamingwiki) are currently not very responsive, their maintainer are busy, and their choice of infra is a bit slow. there are many of their entries that are outdated already. many also require confirmation of what's behind reported.
if they don't choose something more dynamic, like what you provide here, with github and PR, their list will often lack a bit behind.

you tell me if you prefer i do entries from scratch ignoring what ASG and AGW already provide, or if i can do a mix of everything (providing credits when needed too).

**Comment by ThatGuySam** [comment]
So you're thinking store the data in the Github repo in something like JSON or Markdown? 

We could also just hook additional data onto what we already import from ASG via the page slug. 

So we store the reported data with a slug and if that slug matches an existing game then it will display on the respective page.

**Comment by nsklaus** [comment]
i made a confusion, i said "PR", in previous posts, but it should have been "issue" instead. 

i submitted 1 app before, that was brave browser. i thought it was easy and fast to do, there was template available and all. i just had to fill in the data, and that was it. 

so, when submitting an issue, you already have: `App Request Template` and `App Update Template`, if you provided a `game request template`and a `game update template` too, that would be nice.

many game could be added and also kept updated. just like you do with apps, but with games too.

i had no intention of suggesting modification about the way you store data or the format you use for it. 
i just wanted to suggest to extend what you already do for apps with games too.

basicaly: if you made `game request template` and `game update template`, i'd use that. me and of course possibly many other contributors.

**Comment by ThatGuySam** [comment]
That makes sense. 

Normally we store information about apps in the README.md file, however, the format is very specific and limited on what data is displayed, like so: 

```
[App Name](https://downloadwebsite.com) - ✳️ Yes or No, App Status - [Link #1](https://app.com) [Link #2](https://app.com)
```

If we wanted to start storing and providing more data we could store it as files in a folder. 

Homebrew does something similar to this with Casks as Ruby scripts: 
https://github.com/Homebrew/homebrew-cask/tree/master/Casks

And each report could be listed as a markdown file formatted for [gray matter](https://github.com/jonschlinkert/gray-matter#what-does-this-do) parsing, like so: 
```
// batman-arkham-city-nsklaus.md
---
slug: batman-arkham-city
renderingApi: opengl
environment: crossover
supportsControllers: yes
gameVersionTested: 1.0.2
---
## This is the start of my content and this will be shown as the excerpt.
These are my notes about this report
```

**Comment by nsklaus** [comment]
that's more your territory than mine there, since doesitarms is yours. i haven't checked what's going on behind the scene, nor looked at the repo files in details yet, but what you suggest looks correct.. contributor won't see much of it anyway, that's the 'backend' portion, how data is stored and in which format.. as long as we can click 'add issue > game update template > .. and start fill in infos,
i'm all for it :)

**Comment by ThatGuySam** [comment]
Yeah, I could build a tool that will compose the markdown file for you so you could drop it right into the issue you're creating

**Comment by nsklaus** [comment]
about the template, i would like to suggest a few elements that each game entry might need:

title: ... dying light
author / vendor: .... ( link to dev site, GOG, steam, appstore, ..)
status: ... (runs on M1 yes, no, kindof, .. (25 fps minimum to be defined as 'run well' ) )
metacritic score : .....
environment:  ( crossover, wineskin, wine, parallels, rosetta2, native, .. )
technology: (frameworks and apis, such as directx9,10,11, opengl, metal, sdl2, ..)
picture : box art
screenshot: ... ( optional, between 0 to 3  maybe ?  shown as clickable thumbnails )
test machine: ... (mac mini, mbp, mba, ..)
controller: .. (ds4, xbox, steam pad, ..)

install: .. (steps needed to install, if anything special )
settings : .. (steps needed to setup so it run well: example alien isolation: disable AA)
tweaks : .. (tweaks needed if any, example: wine specific configuration for dll native/ builtin ..)
miscellaneous note: .... (optional)

most of this should fit in the same page without even the need to scroll down most of the time. one quick glance you get all info.

**Comment by ThatGuySam** [comment]
Some of those could just be freeform content that appears below, for example: 

```
// batman-arkham-city-nsklaus.md
---
  title: Batman: Arkham City
  slug: batman-arkham-city
  developers:
  - Rocksteady Studios
  - WB Games Montréal
  vendor: Steam
  technologies: 
  - Metal
  - DLSS
  images:
  - https://imgur.com/....
  - https://imgur.com/....
  machine: M1 MacBook Pro
  controller: xbox
  gameVersionTested: 1.0.2
---

## Run Notes
Runs between 25 and 40 fps

## Settings Notes
Ultra everything, 4k, No AA, Motion Blur off

## Tweak Notes
Must run Wine in 32bit mode, etc...

## Other Notes


```


This would allow for adding more custom notes if needed.

**Comment by nsklaus** [comment]
sure, why not, would be great. 
also you included an important point that i forgot about: game version. good one.

**Comment by OliviaDavis14** [comment]
This DoMyEssay.net review opened my eyes. Now I understand that I will not allow myself to be deceived. Besides, I am glad that I learned a lot of new things.
https://onlineessaywritingservice.review/reviews/domyessay-review
